src/status_im/contexts/centralized_metrics/tracking.cljs

```diff
@@ -19,36 +21,31 @@
   [view-id]
   (key-value-event "navigation" :viewId view-id))
 
+(defn screen-event
+  [screen event-data]
+  (let [screen-id (:name screen)
+        view-id   (get-in screen [:metrics :alias-id] screen-id)]
+    {:metric
+     {:eventName  "navigation"
+      :platform   platform/os
+      :appVersion build/app-short-version
+      :eventValue (assoc event-data :viewId (name view-id))}}))
+
 (def ^:const app-started-event "app-started")
 
 (def ^:const view-ids-to-track
   #{;; Tabs
     :communities-stack
     :chats-stack
-    :wallet-stack
-
-    ;; Onboarding
-    :screen/onboarding.intro
-    :screen/onboarding.new-to-status
-    :screen/onboarding.sync-or-recover-profile
-    :screen/onboarding.enter-seed-phrase
-    :screen/onboarding.create-profile
-    :screen/onboarding.create-profile-password
-    :screen/onboarding.enable-biometrics
-    :screen/onboarding.generating-keys
-    :screen/onboarding.enable-notifications
-    :screen/onboarding.preparing-status
-    :screen/onboarding.sign-in-intro
-    :screen/onboarding.sign-in
-    :screen/onboarding.syncing-progress
-    :screen/onboarding.syncing-progress-intro
-    :screen/onboarding.syncing-results
-    :screen/onboarding.welcome})
+    :wallet-stack})
 
 (defn track-view-id-event
   [view-id]
-  (when (contains? view-ids-to-track view-id)
-    (navigation-event (name view-id))))
+  (if-let [screen (get screens/screens-by-name view-id)]
+    (when (get-in screen [:metrics :track?])
```

Comment:
Thank you. This approach is much better than what I did with manually adding the screen names.